ksqldb-execution/src/main/java/io/confluent/ksql/execution/codegen/SqlToJavaVisitor.java

```diff
@@ -448,49 +451,115 @@ private String formatExpression(final Expression expression) {
         final FunctionCall node, final TypeContext context
     ) {
       final FunctionName functionName = node.getName();
-
       final String instanceName = funNameToCodeName.apply(functionName);
 
-      final UdfFactory udfFactory = functionRegistry.getUdfFactory(node.getName());
-
-      // this context gets updated as we process non lambda arguments
-      final TypeContext currentTypeContext = context.getCopy();
-
       final List<SqlArgument> argumentSchemas = new ArrayList<>();
       final List<TypeContext> typeContextsForChildren = new ArrayList<>();
-      final boolean hasLambda = node.hasLambdaFunctionCallArguments();
+      final List<SqlArgument> argTypesForFirstPass = new ArrayList<>();
 
-      for (final Expression argExpr : node.getArguments()) {
-        final TypeContext childContext = TypeContextUtil.contextForExpression(
-            argExpr, context, currentTypeContext
-        );
-        typeContextsForChildren.add(childContext);
-
-        // pass a copy of the context to the type checker so that type checking in one
-        // expression subtree doesn't interfere with type checking in another one
-        final SqlType resolvedArgType =
-            expressionTypeManager.getExpressionSqlType(argExpr, childContext.getCopy());
-        if (argExpr instanceof LambdaFunctionCall) {
-          argumentSchemas.add(
+      final List<Expression> arguments = node.getArguments();
+
+      for (final Expression expression : arguments) {
```

Comment:
Not for this PR, but I'm feeling a bit harder for remembering the resolved types now since we are making two passes.. :)

Reply:
hopefully the new refactor into the utility class makes it clearer